Sources/SuperwallKit/Misc/Extensions/UIWindow/UIWindow+SwizzleSendEvent.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  File.swift
+//  
+//
+//  Created by Yusuf Tör on 01/09/2023.
+//
+
+import UIKit
+
+extension UIWindow {
+  /// Does a switcharoo with the UIWindow's `sendEvent` method and our own method so that
+  /// we can intercept the first `began` touch event.
+  static func swizzleSendEvent() {
+    let originalSelector = #selector(UIWindow.sendEvent(_:))
+    let swizzledSelector = #selector(swizzledSendEvent(_:))
+
+    guard
+      let originalMethod = class_getInstanceMethod(UIWindow.self, originalSelector),
+      let swizzledMethod = class_getInstanceMethod(UIWindow.self, swizzledSelector)
+    else {
+      return
+    }
+
+    method_exchangeImplementations(originalMethod, swizzledMethod)
+  }
+
+  /// Tracks a `.touchesBegan` event for the first `began` touch event received on the `UIWindow`.
+  @objc private func swizzledSendEvent(_ event: UIEvent) {
+    if event.type == .touches {
+      // Check for a began touch event.
+      guard event.allTouches?.filter({ $0.phase == .began }) != nil else {
```

Comment:
Would an array that only has touches with .end filter down to an empty array (instead of nil) and unintentionally continue past the guard?

Reply:
You're totally right